docs/chain/did.md

```diff
@@ -0,0 +1,73 @@
+
+A full DID is needed if the user wants to become an attester or wants to setup delegations. A full DID also allows the user to embed a list of URLs, known as service endpoints, into the DID document so that it can be retrieved from the chain.
+
+To create a full DID the user first has to create a set of keys and service endpoints:
+
+* one authentication key for signing extrinsics from your DID
+* zero or more key agreement keys for encrypting messages that are sent to you
+* one attestation key for signing attestations (optional)
+* one delegation key for authorizing delegations (optional)
+* service endpoints that point to external hostings for others to find (optional)
+
+With these they are ready to write the DID to the KILT blockchain.
+
+The user then has to create the `did::create` extrinsic and sign it with any KILT account that has enough funding to pay both the transaction fees and the  DID deposit. The extrinsic consists of  
+
+* The `DidCreationDetails` object containing keys and service endpoints
+* The `DidSignature` which is a signature using your authentication key over the scale encoded `DidCreationDetails` from above
+* A regular signature authenticating the sender of the extrinsic
+
+Once this extrinsic is submitted and executed, the DID is written to the chain.
+
+## Updating a DID
+
+There is a whole set of extrinsics needed to update a full DID. These are:
+
+* `set_authentication_key`
+* `set_delegation_key`
+* `remove_delegation_key`
+* `set_attestation_key`
+* `remove_attestation_key`
+* `add_key_agreement_key`
+* `remove_key_agreement_key`
+* `add_service_endpoint`
+* `remove_service_endpoint`
+* `delete`
+
+
+All of them have to be authenticated using the DID that is updated. For that there is the special `submit_did_call` extrinsic. This extrinsic contains another call and a signature from the DID that is authorizing the inner call.
+
+For example when you want to add a new service endpoint:
+
+1. Create a `DidEndpoint` object containing a service id, a list of service types and a list of URLs.
+2. Wrap it in an `add_service_endpoint` extrinsic.
+3. Take this call and put it into a `DidAuthorizedCallOperation` object together with the DID, the tx_counter of the DID, the current block number and the account id of the submitter of the final extrinsic.
+4. Now we can put together the actual `submit_did_call` extrinsic by signing the scale encoded `DidAuthorizedCallOperation` with our DID and then signing and submitting the call together with the DID-signature using the key of the account that is specified as `submitter` in the `DidAuthorizedCallOperation`.
+
+The extrinsic will then be dispatched to the `did::submit_did_call` function which will verify that 1) the submitters signature is correct and 2) that the DID signature is correct. After that it will execute the inner call with a special `DidOrigin` that allows the inner call to access both the account id of the submitter and the id of the affected DID.
```

Comment:
```suggestion
The `did::submit_did_call` will verify that 1) the submitters signature is correct and 2) that the DID signature is correct. After that it will execute the inner call with a special `DidOrigin` that allows the inner call to access both the account id of the submitter and the id of the affected DID.
```

I hope there are more than those two checks in the code. 😁 The `tx_counter` and block number are there for a reason I suppose.